pkg/chunk/chunk_store.go

```diff
@@ -118,29 +134,25 @@ func (c *store) Put(ctx context.Context, chunks []Chunk) error {
 
 // PutOne implements ChunkStore
 func (c *store) PutOne(ctx context.Context, from, through model.Time, chunk Chunk) error {
-	userID, err := user.ExtractOrgID(ctx)
+	// Encode the chunk first - checksum is calculated as a side effect.
+	buf, err := chunk.Encode()
 	if err != nil {
 		return err
 	}
+	batch := c.storage.NewWriteBatch()
+	batch.AddChunk(c.storage, chunk, buf)
+	c.writer.Write <- batch
```

Comment:
This is assuming that the chunks are put in a table-like store - how would this work with S3 (and the upcoming GCS?)

Reply:
Also, having the interface to a Writer be a put to a channel seems unfriendly; perhaps this should be a function?